rubi/rubi/contracts/erc20.py

```diff
@@ -1,147 +1,43 @@
+    """this class represents a contract that implements the ERC20 standard.
 
     :param w3: Web3 instance
     :type w3: Web3
     :param contract: Contract instance
     :type contract: Contract
-    :param wallet: a wallet address of the signer (optional, default is None)
-    :type wallet: Optional[ChecksumAddress]
-    :param key: the private key of the signer (optional, default is None)
-    :type key: Optional[str]
     """
 
     def __init__(
         self,
         w3: Web3,
         contract: Contract,
-        contract_type: ContractType = ContractType.ERC20,
-        wallet: Optional[ChecksumAddress] = None,
-        key: Optional[str] = None,
     ):
         """constructor method"""
         super().__init__(
             w3=w3,
             contract=contract,
-            contract_type=ContractType.ERC20,
-            wallet=wallet,
-            key=key,
-        )
-
-        self.name: str = self.name()
-        self.symbol: str = self.symbol()
-        self.decimal: int = self.decimals()
-
-    @classmethod
-    def from_network(
-        cls,
-        name: str,
-        network: Network,
-        wallet: Optional[ChecksumAddress] = None,
-        key: Optional[str] = None,
-    ) -> "ERC20":
-        """Create an ERC20 instance based on a Network instance and token name.
-
-        :param name: The name of the token.
-        :type name: str
-        :param network: A Network instance.
-        :type network: Network
-        :param wallet: Optional wallet address to use for interacting with the contract (optional, default is None).
-        :type wallet: Optional[ChecksumAddress]
-        :param key: Optional private key for the wallet (optional, default is None).
-        :type key: Optional[str]
-        :return: An ERC20 instance based on the Network instance and token name.
-        :rtype: ERC20
-        :raises Exception: If the token name does not exist in the network configuration.
-        :raises Exception: If the ERC20.json ABI file is not found in the network_config folder.
-        """
-
-        if network.token_addresses[name] is None:
-            raise Exception(
-                f"{name} in not a valid token according to the network config."
-            )
-
-        abi: ABI
-
-        try:
-            path = f"{os.path.dirname(os.path.abspath(__file__))}/../../network_config/ERC20.json"
-
-            with open(path) as f:
-                abi = json.load(f)
-
-        except FileNotFoundError:
-            raise Exception(
-                "ERC20.json abi not found. this file should in the network_config folder"
-            )
-
-        return cls.from_address_and_abi(
-            w3=network.w3,
-            address=network.token_addresses[name],
-            contract_abi=abi,
-            contract_type=ContractType.ERC20,
-            wallet=wallet,
-            key=key,
         )
 
-    @classmethod
-    def from_address(
```

Comment:
i think we will want to keep a `@classmethod` that allows the user to pass in an address to create an `ERC20` class. this will allow us to easily utilize this functionality with the `Network` object to add new assets to the `Client`

Reply:
i say this, and come back to the conversation we had a while ago where we introduced the `custom_token_addresses_file`. 

this goes two ways: 1) adding guard rails to require the user to stop the program to add new assets, 2) letting the user have the optionality to add assets as the program runs. 

i think i lean towards option 2, we can keep from exposing this at the `client` level to attempt to bias towards the `custom_token_addresses_file`

Reply:
@denverbaumgartner on this the `from_address` method has been preserved and just moved to the `BaseContract` so now it can be called for any `ERC20` and additionally for the `RubiconMarket` and `RubiconRouter` classes.